service/dap/server.go

```
@@ -1049,6 +1071,32 @@ func (s *Session) onLaunchRequest(request *dap.LaunchRequest) {
 		return
 	}
 
+	func() {
+		if args.Console == "internalConsole" && runtime.GOOS == "linux" && (args.Backend == "default" || args.Backend == "native") {
```

It's weird that using internalConsole on an unsupported configuration fails silently. It's also weird that all values of console that aren't internalConsole are accepted as valid.

---

The current behavior is that `console` is not checked, so it can also be completely invalid and `dlv dap` would accept it, so I'm not sure we necessarily have to validate the `console` argument.

---

Suppose someone misspells internlaConsole, how do they notice?

---

Usually, such syntax can be checked in VSCode itself - it can present an error such as: `"internalConsole" is not a valid "console" value".

---

Not everyone uses vscode with dap. I thought that was the whole point of this.

---

Hm, true. I am not familiar with how every DAP client handles this, but adding an allowed list of declared DAP consoles (I think it is standardized) should be easy enough.

---

it should just be the consoles that we support.